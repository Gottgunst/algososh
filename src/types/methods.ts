export enum Methods {
  addHead = 'add-head',
  addTail = 'add-tail',
  addIndex = 'add-index',
  delHead = 'del-head',
  delTail = 'del-tail',
  delIndex = 'del-index',
}
