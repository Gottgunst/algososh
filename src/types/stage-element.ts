export type TStageElement<S> = (props: TStageElementProps<S>) => JSX.Element;
type TStageElementProps<S> = {
  stages: S;
  lap: number;
  phase?: 'initial' | 'animate' | 'finally';
};
