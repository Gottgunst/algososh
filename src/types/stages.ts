export type stages<T> = T[][];
