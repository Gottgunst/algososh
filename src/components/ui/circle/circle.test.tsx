import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

// Проверяем при помощи снэпшотов корректную отрисовку элемента:
describe('Circle', () => {
  // без буквы;
  it('Render without letter', () => {
    //############################## arrange
    const circle = renderer.create(<Circle />).toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с буквами;
  it('Render with letter', () => {
    //############################## arrange
    const circle = renderer.create(<Circle letter='A' />).toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с head;
  it('Render with head', () => {
    //############################## arrange
    const circle = renderer.create(<Circle letter='A' head='Top' />).toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с react-элементом в head;
  it('Render with head Node', () => {
    //############################## arrange
    const headNode = <a href='ya.ru'>42</a>;
    const circle = renderer
      .create(<Circle letter='A' head={headNode} />)
      .toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с tail;
  it('Render with tail', () => {
    //############################## arrange
    const circle = renderer.create(<Circle letter='A' tail='Top' />).toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с react-элементом в tail;
  it('Render with tail Node', () => {
    //############################## arrange
    const tailNode = <a href='ya.ru'>42</a>;
    const circle = renderer
      .create(<Circle letter='A' tail={tailNode} />)
      .toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с index;
  it('Render with index', () => {
    //############################## arrange
    const circle = renderer.create(<Circle letter='A' index={42} />).toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // с пропом isSmall ===  true;
  it('Render isSmall === true', () => {
    //############################## arrange
    const circle = renderer.create(<Circle letter='A' isSmall />).toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // в состоянии default;
  it('Render state — Default', () => {
    //############################## arrange
    const circle = renderer
      .create(<Circle letter='A' state={ElementStates.Default} />)
      .toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // в состоянии changing;
  it('Render state — Changing', () => {
    //############################## arrange
    const circle = renderer
      .create(<Circle letter='A' state={ElementStates.Changing} />)
      .toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });

  // в состоянии modified.
  it('Render state — Modified', () => {
    //############################## arrange
    const circle = renderer
      .create(<Circle letter='A' state={ElementStates.Modified} />)
      .toJSON();
    //############################## assert
    expect(circle).toMatchSnapshot();
  });
});
