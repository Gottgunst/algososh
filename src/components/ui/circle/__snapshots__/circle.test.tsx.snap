// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Render isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render state — Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render state — Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render state — Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Top
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render with head Node 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <a
      href="ya.ru"
    >
      42
    </a>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    42
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle Render with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Render with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Top
  </div>
</div>
`;

exports[`Circle Render with tail Node 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <a
      href="ya.ru"
    >
      42
    </a>
  </div>
</div>
`;

exports[`Circle Render without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
